### PR TITLE
Use opentype.js to test font data

### PR DESCRIPTION
**Pull request by yisibl**
Through this PR, we can better test fonts and no longer use binary-based comparison methods.

* This PR incidentally fixes all options must remove extra white space.
* This will help improve the https://github.com/fontello/svg2ttf/pull/86 test.

### CODE REVIEW

**Comment by yisibl** [comment]
@puzrin PTAL.

**Comment by yisibl** [comment]
@puzrin We really need to test based on the font table, I recently upgraded from v2.0.0 to the latest version and there were a lot of problems. Without these tests, it is difficult to troubleshoot and find problems.

File: index.js [hunk 1]
```diff
@@ -28,10 +28,10 @@ function svg2ttf(svgString, options) {
   font.copyright = options.copyright || svgFont.metadata;
   font.description = options.description || 'Generated by svg2ttf from Fontello project.';
   font.url = options.url || 'http://fontello.com';
-  font.sfntNames.push({ id: 2, value: options.subfamilyname || svgFont.subfamilyName || 'Regular' }); // subfamily name
-  font.sfntNames.push({ id: 4, value: options.fullname || svgFont.id }); // full name
+  font.sfntNames.push({ id: 2, value: (options.subfamilyname || svgFont.subfamilyName || 'Regular').trim() }); // subfamily name
```
Comment (puzrin) [inline index.js:31]: I don't like idea of auto-trim. IMO that's a garbage for artificial case.

**Comment by yisibl** [comment]
Sorry, please let me test in the fork's `svg2ttf-new` first. I need to solve the problem of height in Windows first. Later, I will deploy to our product to test, if there is no problem, I will clean up this PR.

**Comment by puzrin** [comment]
@yisibl see master updates. IMO this PR has no more worth to merge.

**Comment by yisibl** [comment]
> @yisibl see master updates. IMO this PR has no more worth to merge.

I can resubmit a new PR.

File: index.js (hunk 1 above)
Comment (yisibl) [reply index.js:31]: I prefer svg2ttf to filter out these extraneous whitespace characters in order to produce a cleaner font file. Similar processing is also done in ufo2ft, of course we can only consider familyName and styleName.

https://github.com/googlefonts/ufo2ft/blob/a5267d135d5a8dba7e6a5d3ac44139afa6159429/Lib/ufo2ft/fontInfoData.py#L74

**Comment by puzrin** [comment]
@yisibl if you think something important was missed in master commits - please do PR, i will review.

File: index.js (hunk 1 above)
Comment (puzrin) [reply index.js:31]: Sorry, i do not agree, IMO tool should write data exactly as been passed. IMO if you need trimming, it should be filtered on upper app level.

**Comment by puzrin** [comment]
IMO master is ready for release, but i'll be waiting for your comments for sure.

File: index.js (hunk 1 above)
Comment (yisibl) [reply index.js:31]: I also changed my mind, and now I agree with you.

**Pull request closed without merging**